### DISCUSSION TITLE
Port Forwarding with Ngrok – Setting Up and Troubleshooting

### DISCUSSION THREAD

**Discussion by ab-h-i-n** [pocketbase/pocketbase#6013] [Q&A]
I'm currently using Ngrok to expose my local server (e.g., PocketBase) to the internet, and I'm having some difficulties with port forwarding. I am trying to forward traffic from a local server running on port 8090 using Ngrok, but I am encountering the following issue:

Error Message: ERR_NGROK_8012: Traffic was successfully tunneled to the ngrok agent, but the agent failed to establish a connection to the upstream web service at http://localhost:8090. The error encountered was: dial tcp [::1]:8090: connect: connection refused.

**Comment by ganigeorgiev** [comment]
Double check that the PocketBase executable is actually running and it is accessible.

Testing locally with `ngrok http 8090` works correctly for me.

It could be a firewall issue or something else OS related (I'm testing on Linux/Fedora with ngrok 3.18.4).

Since the localhost resolution also depends on your /etc/hosts config, you can try to specify explicitly the host as part of the ngrok command, aka. `ngrok http 127.0.0.1:8090` so that it resolves to `http://127.0.0.1:8090`.

**Reply by ab-h-i-n** [reply]
![image](https://github.com/user-attachments/assets/9a5e6c70-f055-47cd-8c4a-98f2008e725b)
![image](https://github.com/user-attachments/assets/8f302993-9f2c-4db2-9db0-515a77bc7d22)


Can you help me to find what causing this error. i have hosted it in aws ec2
